Swift/Sources/StateMachineMacros/Macros/StateMachineHashableMacro.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  Copyright (c) 2024, Match Group, LLC
+//  BSD License, see LICENSE file for details
+//
+
+import SwiftSyntax
+import SwiftSyntaxMacros
+
+public struct StateMachineHashableMacro: ExtensionMacro {
+
+    public static func expansion(
+        of node: AttributeSyntax,
+        attachedTo declaration: some DeclGroupSyntax,
+        providingExtensionsOf type: some TypeSyntaxProtocol,
+        conformingTo protocols: [TypeSyntax],
+        in context: some MacroExpansionContext
+    ) throws -> [ExtensionDeclSyntax] {
+
+        guard let enumDecl: EnumDeclSyntax = declaration.as(EnumDeclSyntax.self)
+        else { throw StateMachineHashableMacroError.typeMustBeEnum }
+
+        let elements: [EnumCaseElementSyntax] = enumDecl
+            .memberBlock
+            .members
+            .compactMap { $0.as(MemberBlockItemSyntax.self) }
+            .map(\.decl)
+            .compactMap { $0.as(EnumCaseDeclSyntax.self) }
+            .flatMap(\.elements)
+
+        guard !elements.isEmpty
+        else { throw StateMachineHashableMacroError.enumMustHaveCases }
+
+        let enumCases: [String] = elements
+            .map(\.name.text)
+            .map { "case \($0)" }
+
+        let hashableIdentifierCases: [String] = elements
+            .map(\.name.text)
+            .map { "case .\($0):\nreturn .\($0)" }
+
+        var associatedValueCases: [String] = []
+        for element: EnumCaseElementSyntax in elements {
+            if let parameters: EnumCaseParameterListSyntax = element.parameterClause?.parameters, !parameters.isEmpty {
+                if parameters.count > 1 {
```

Comment:
I think this same code could be used for both the `parameters.count == 1` case and the `parameters.count > 1 case`. If parameters.count == 1, then the range will be 1...1 (just 1) and there will just be a single value. So you might not need the if / else here, just this "if" implementation.

Reply:
You are correct 👍

Was only implemented this way so the name of a single associated value is `value` instead of `value1`. While I understand that this is generated code and therefore `value1` is entirely appropriate, I feel there is no harm in leaving the current implementation.

Reply:
Ah, ok 👍